### ISSUE TITLE
Audiosurf (12900)

### ISSUE THREAD

**Issue by pprotonrep** [ValveSoftware/Proton#44]
I tested [Audiosurf](https://store.steampowered.com/app/12900/) and it worked perfectly fine for me, including sounds, controller input and effects.

- Ubuntu 18.04.1 LTS (64 bit 4.15.0-32-generic)
- GeForce GTX 1070/PCIe/SSE2 (396.51)
- Proton 3.7 (default configuration)

[System information from steam](https://gist.github.com/pprotonrep/43f7265a7457b17ceb5b848e89c26147)

**Comment by daniel-j** [comment]
Crashes at launch for me. I can see the process running, but no window is created.

* Ubuntu 18.04.1 (Linux 4.16 64 bit)
* Mesa DRI Intel HD Graphics
* Proton 3.7

EDIT: It runs now. All I did was reinstall it. Weird!
EDIT2: It crashes on my main PC, see my other comment.

**Comment by valeth** [comment]
One issue I noticed with this game is that some characters in filenames aren't rendered.
Not sure if this is actually an issue with the game itself.
Can anyone confirm this on native Windows?

**Comment by ArekPiekarz** [comment]
It crashes for me even after reinstallation. A window with loading appears, the progress bar fills up halfway and then it exits the game.

Config:
Xubuntu 18.04 x64
Geforce GTX 1050 Ti, Nvidia drivers 396.51

[System info](https://gist.github.com/ArekPiekarz/89c0066a2c86b4d6752c0460b7add468)

Edit: Added logs from console:
[audiosurf-logs.txt](https://github.com/ValveSoftware/Proton/files/2315426/audiosurf-logs.txt)

**Comment by commandhat** [comment]
@valeth Correct. Audiosurf does not render certain symbols in filenames on native Windows.

**Comment by daniel-j** [comment]
I get a crash on my main PC. I see a progress bar filling up half way and then it crashes. Wine dump:
https://pastebin.com/raw/f4YbMKQZ

Kernel: 4.17.14-arch1-1-ARCH
OS: Arch Linux
GPU: Nvidia GTX 980 Ti
CPU: Intel i7-6700k

EDIT: Installing `ttf-ms-fonts` package from AUR fixes it. The game seems to require some fonts or else it crashes.

**Comment by ArekPiekarz** [comment]
After installing MS fonts* Audiosurf works correctly now, the performance is great. Tested on just released Nvidia drivers 396.54.

The only bugs I noticed were:
- the tutorial video does not play when clicked on it
- the road is bright white with graphics set to "Premium" and background set to black in the first run of tutorial; after it the glitch disappeared, even when checking the tutorial the second time 
- some text is overlapping itself, like in tutorial "Overfilled" has the first and the second letter in the same place or after finishing a race on the scoreboard there is "Other bonuses" with "O" and "t" overlapping

Note that I haven't checked if these bugs are present on Windows yet.

--------------

\* For installing fonts on Ubuntu this command didn't work:
`sudo apt install ttf-mscorefonts-installer`
because of this bug: https://bugs.launchpad.net/ubuntu/+source/msttcorefonts/+bug/1713615, but these did the trick:
```
wget http://ftp.de.debian.org/debian/pool/contrib/m/msttcorefonts/ttf-mscorefonts-installer_3.7_all.deb
sudo apt install ./ttf-mscorefonts-installer_3.7_all.deb
```

**Comment by ghost** [comment]
Worked fine out of the box for me (Debian Buster, Intel i3-6100, GTX970), with some slight issues with font rendering in menus (the `ttf-mscorefonts-installer` package was already installed).

EDIT - [SysInfo](https://gist.github.com/DrWindu/1a10ea5e4305d2f4716bc75fdf6fdfff)
EDIT 2 - Audiosurf players might want to symlink their music library to easily access it in game, with something like (depending on your folder hierarchy) :
```
cd .steam/steam/steamapps/compatdata/12900/pfx/drive_c/users/steamuser/
rmdir "My Music"
ln -s $HOME/Music "My Music"
```

**Comment by derram** [comment]
![audiosurf_434](https://user-images.githubusercontent.com/5730335/44629539-7f9a6680-a916-11e8-92c8-f3533cfc0be0.png)

Game is missing a lot of textures for me, and playing it on "premium" graphical settings results in a completely black screen.

**Comment by ArekPiekarz** [comment]
@derram Can you copy the info from Steam in Help -> System Information to Gist and link it here?

**Comment by derram** [comment]
@ArekPiekarz https://gist.github.com/derram/50bf1eb3e50a426b63505366e2a45def

**Comment by DanySK** [comment]
Game starts when installing the package suggested by @daniel-j . Performance and graphics (on premium) looks ok, apart from font rendering, that looks horrible.

**Comment by AwesamLinux** [comment]
- GPU: RX VEGA 56
- Driver/LLVM version: Mesa 19.2.0-devel/9.0.0
- Kernel version: 5.1.2-050102-lowlatency
- Distro version: Linux Mint 19.1 Tessa (64 bit) (Cinnamon Desktop)
- Proton version: 4.2-4
[system_info.txt](https://github.com/ValveSoftware/Proton/files/3192600/system_info.txt)

Game worked fine but the Tutorial videos do not play.

[media_info.txt](https://github.com/ValveSoftware/Proton/files/3192601/media_info.txt)
[steam-12900.log](https://github.com/ValveSoftware/Proton/files/3192602/steam-12900.log)

In the log I spotted "err:quartz:GetClassMediaFile Media class not found"

#1464 game uses quartz and the videos are WMV, VC-1 + WMA

✔ Switching between desktop and game works
✔ Nintendo Switch Pro Controller works

**Comment by Obbi89** [comment]
Hello. The game freezes after some time (for some time) repeatedly on Proton > 7

**Comment by waspennator** [comment]
[steam-12900.zip](https://github.com/user-attachments/files/16861358/steam-12900.zip)


Tried it on my oled, I can confirm that despite being verified on deck, the game has moments during songs where it'll freeze for a certain amount of seconds before suddenly speeding up rapidly. So far at least on my end, I've only been able to replicate this on 9.0-2 and Experimental.

**Comment by alasky17** [comment]
@Obbi89 @waspennator In our testing, the game is playing well with wine new media source in current experimental.  Please let me know if you are still able to reproduce any freezing with Proton Experimental :)

**Comment by waspennator** [comment]
Might have to test some more to make sure, but i retried still alive on proton Experimental and I don't believe i noticed the same lengthy freezing that was present on older proton versions

**Comment by Obbi89** [comment]
I'm not sure if I did test this on my rig and steam deck last time but I do believe it was causing issues either way.
I'll just did some testing on the deck and it seems to work now :)
Thanks!
